internal/cache/cache.go

```diff
@@ -0,0 +1,110 @@
+package cache // import "gopkg.in/juju/charmstore.v5-unstable/internal/cache"
+
+import (
+	"math/rand"
+	"sync"
+	"time"
+
+	"gopkg.in/errgo.v1"
+)
+
+type entry struct {
+	value  interface{}
+	expire time.Time
+}
+
+type Cache struct {
+	mu sync.Mutex
+
+	// expire holds when the cache is due to expire.
+	expire   time.Time
+	maxAge   time.Duration
+	old, new map[string]entry
```

Comment:
Not sure about using "new" as an attribute, it can be confusing, it apparently confuses github syntax highlighting ;-)
[shrug]